components/charts/bar.js

```diff
@@ -0,0 +1,80 @@
+import { useState } from "react";
+import { db } from "../../firebaseConfig";
+import { Bar } from "react-chartjs-2";
+
+export default function BarGraph(props) {
+  const [data, setData] = useState({
+    val: [],
+    label: [],
+    color: [],
+  });
+
+  db.collection("1A Data")
+    .doc(props.datatype)
+    .onSnapshot(
+      async (snapshot) => {
+        let data = {
+          val: [],
+          label: [],
+          color: [],
+          title: "",
+        };
+
+        await snapshot.data().x.forEach((element) => {
+          data.val.push(element.value);
+          data.label.push(element.index);
+          data.color.push(element.color);
+        });
+        data.title = snapshot.data().title;
+        setData(data);
+      },
+      (err) => {
+        console.log("Error fetching firebase snapshot! " + err);
+      }
+    );
+
+  return (
+    <div>
+      {/* <div className="header">
+        <h1 className="title">Doughnut</h1>
+      </div> */}
+      <div className="chart">
+        <Bar
+          data={{
+            labels: data.label,
+            datasets: [
+              {
+                label: "# of Students",
```

Comment:
is this supposed to be a generic component? if it is, this should be based on data in firebase

Reply:
label***

Reply:
I dont think it will always be # of students rii so